vcd/resource_vcd_vapp_vm.go

```diff
@@ -1064,6 +1079,12 @@ func createVmFromTemplate(d *schema.ResourceData, meta interface{}, vmType typeO
 		return nil, fmt.Errorf("error refreshing VM %s : %s", vmName, err)
 	}
 
+	// Handle VM Security Tags settings
+	err = createOrUpdateVmSecurityTags(d, vm)
+	if err != nil {
+		return nil, fmt.Errorf("[VM create] error creating security tags for VM %s : %s", vm.VM.Name, err)
+	}
+
```

Comment:
In fact, this one and https://github.com/vmware/terraform-provider-vcd/pull/1006/files#diff-d888384050e0ea29e45bdfbcd476b8c09c90ec98ab5c9bfbcfe5491760497847R1425-R1430 are not needed. 

The way it works is `genericResourceVmCreate` conditionally calls `createVmFromTemplate` or `createVmEmpty` based on the type of VM and then finalizes by performing "common" actions for all VMs. Security tags are common and it is sufficient to have it here https://github.com/vmware/terraform-provider-vcd/pull/1006/files#diff-d888384050e0ea29e45bdfbcd476b8c09c90ec98ab5c9bfbcfe5491760497847R801-R808

Reply:
I had a hunch that it should be sufficient to put it into `genericResourceVmCreate `, but I was not sure. Thanks for clarifying this :)